### ISSUE TITLE
IDE0078 - Use pattern matching suggests incorrect code when comparing to constants converted with user-defined casts

### ISSUE THREAD

**Issue by Blokyk** [dotnet/roslyn#66787]
**Version Used**:  `4.5.0-2.22527.10 (401447a2)`

**Steps to Reproduce**:

```csharp
Int128 val = Foo();

// IDE0078: Use pattern matching
if (val > Int32.MaxValue || val < Int32.MinValue) {...}

// Suggested codefix:
if (val is > (Int128)Int32.MaxValue or < (Int128)Int32.MinValue) {...} // CS0150: A constant value is expected
```

Here's a [Sharplab link](https://sharplab.io/#v2:EYLgtghglgdgPgAQEwEYCwAoTBJGAXFJADgAIA3CAGxIF4SAxAe0YAoBKAbk0wHoeTsAEQCiABlEB2IiBIBVAM4BTEgAcIePIoBOMEpDwBjABawA5pigAzEiwrUAfCVh4AdAFkIADwBqVAK7KcHDkVCQAPE747rC+lAFsJADeAL7cGLgExAzM7LSOKBxAA==), although it's not that useful since IDE0078 doesn't seem to show up there.

**Diagnostic Id**: IDE0078

**Expected Behavior**:
No diagnostic is reported, since there's no way to cleanly rewrite the initial condition to use pattern matching.

**Actual Behavior**:
Diagnostics `IDE0078` is reported on the whole condition, and when the codefix is applied, `CS0150` is reported on both cast expressions.
